Clear the alt-svc cache on session end, to prevent potential fingerprinting
---
diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index c35867b90d..33ea20930c 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -9,6 +9,7 @@
 
 #include "ASpdySession.h"
 #include "AltServiceChild.h"
+#include "CacheObserver.h"
 #include "EventTokenBucket.h"
 #include "HttpLog.h"
 #include "LoadContextInfo.h"
@@ -2331,6 +2332,11 @@ nsHttpHandler::Observe(nsISupports* subject, const char* topic,
     // clear cache of all authentication credentials.
     mAuthCache->ClearAll();
     mPrivateAuthCache->ClearAll();
+
+    if (CacheObserver::ClearCacheOnShutdown() && mAltSvcCache) {
+      mAltSvcCache->ClearAltServiceMappings();
+    }
+
     if (mWifiTickler) mWifiTickler->Cancel();
 
     // Inform nsIOService that network is tearing down.
